packages/ember-glimmer/lib/ember-template-compiler/plugins/transform-has-block-syntax.js

```diff
@@ -0,0 +1,58 @@
+/**
+ @module ember
+ @submodule ember-glimmer
+*/
+
+/**
+  A Glimmer2 AST transformation that replaces all instances of
+
+  ```handlebars
+ {{hasBlock}}
+  ```
+
+  with
+
+  ```handlebars
+ {{has-block}}
+  ```
+
+  @private
+  @class TransformHasBlockSyntax
+*/
+
+export default function TransformHasBlockSyntax() {
+  // set later within Glimmer2 to the syntax package
+  this.syntax = null;
+}
+
+const OLD_HAS_BLOCK = 'hasBlock';
+const NEW_HAS_BLOCK = 'has-block';
+
+/**
+  @private
+  @method transform
+  @param {AST} ast The AST to be transformed.
+*/
+TransformHasBlockSyntax.prototype.transform = function TransformHasBlockSyntax_transform(ast) {
+  let { traverse, builders: b } = this.syntax;
+
+  traverse(ast, {
+    PathExpression(node) {
+      if (node.original === OLD_HAS_BLOCK) {
+        return b.sexpr(b.path(NEW_HAS_BLOCK));
```

Comment:
We should pass through the `loc` of the PathExpression into the SubExpression. Even though it is kind of hacky, it still represents the source code that ultimately generated the subexpression. Same applies below in the SubExpression visitor.